pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java

```diff
@@ -560,24 +565,10 @@ protected FunctionDetails convert(FunctionConfig functionConfig)
                 sourceSpecBuilder.setTopicsPattern(functionConfig.getTopicsPattern());
             }
 
-            // Set subscription type based on processing semantics
-            if (functionConfig.getProcessingGuarantees() != null) {
```

Comment:
I don't think we should get rid of this logic.  It is a better user experience if the user just specify a processing guarantee and subscription is automatically set for them.  In general, users will be more familiar with processing guarantee terminology than subscription terminology.  We can have an option for the to override the subscription type from the CLI if the user wants to do that

Reply:
>  It is a better user experience if the user just specify a processing guarantee and subscription is automatically set for them

Processing-guarantee generally implies how to ack the message and retain-ordering describes how to consume message. Therefore, if processing-guarantee is `ATMOST_ONCE ` it means, we want to ack the message immediately no-matter what result is and `ATMOST_ONCE` can be used with SHARED or FAILOVER subscription both (means with or without ordering feature). So, generally processing-guarantee and ordering play separate role in terms of consuming and processing the message.

> If a user specifies a processing guarantee of effectively once, the subscription by default should be Failover

hmm.. yes, that make sense. I will make that change.

Reply:
@jerrypeng  fixed it.